pkg/metrics/check_metrics.go

```diff
@@ -0,0 +1,158 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/ckey"
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+)
+
+var (
+	tlmCheckMetricsTotal   = telemetry.NewGauge("check_metrics", "total_metrics", []string{"stateful"}, "Gauge of all check metrics")
+	tlmCheckMetricsActive  = telemetry.NewGauge("check_metrics", "active_metrics", []string{"stateful"}, "Gauge of non-expired check metrics (total - waiting)")
+	tlmCheckMetricsWaiting = telemetry.NewGauge("check_metrics", "waiting_metrics", []string{"stateful"}, "Gauge of expired check metrics waiting for timeout")
+	tlmCheckMetricsAdded   = telemetry.NewCounter("check_metrics", "created_total", []string{"stateful"}, "Count of new check metrics added")
+	tlmCheckMetricsExpired = telemetry.NewCounter("check_metrics", "expired_total", []string{"stateful"}, "Count of expired metrics")
+	tlmCheckMetricsRemoved = telemetry.NewCounter("check_metrics", "removed_total", []string{"stateful"}, "Count of removed metrics")
+
+	checkMetricsAddSampleTelemetry = &AddSampleTelemetry{
+		Total:     tlmCheckMetricsAdded.WithValues("sum"),
+		Stateful:  tlmCheckMetricsAdded.WithValues("true"),
+		Stateless: tlmCheckMetricsAdded.WithValues("false"),
+	}
+	tlmCheckMetricsExpiredTotal     = tlmCheckMetricsExpired.WithValues("sum")
+	tlmCheckMetricsExpiredStateful  = tlmCheckMetricsExpired.WithValues("true")
+	tlmCheckMetricsExpiredStateless = tlmCheckMetricsExpired.WithValues("false")
+	tlmCheckMetricsRemovedTotal     = tlmCheckMetricsRemoved.WithValues("sum")
+	tlmCheckMetricsRemovedStateful  = tlmCheckMetricsRemoved.WithValues("true")
+	tlmCheckMetricsRemovedStateless = tlmCheckMetricsRemoved.WithValues("false")
+)
+
+// CheckMetrics stores metrics for the check sampler.
+//
+// This is similar to ContextMetrics, but provides additional facility to remove metrics.
```

Comment:
It may be worth defining "expire" and "remove" here, as the distinction is critical but the words don't give the full meaning.